include/bout/region.hxx

```diff
@@ -546,6 +546,18 @@ public:
     indices = getRegionIndices();
   };
 
+  bool operator==(const Region<T>& other) {
+    if (this->size() != other.size()) {
+      return false;
+    }
+    for (auto i1 = this->begin(), i2 = other.begin(); i1 != this->end(); ++i1, ++i2) {
+      if (i1 != i2) {
+        return false;
+      }
+    }
+    return true;
```

Comment:
```suggestion
    return std::equal(begin(), end(), other.begin());
```

Should probably be implemented as a free function rather than a member

Reply:
std::equal in the 3 argument version is actually missing the size check:
https://en.cppreference.com/w/cpp/algorithm/equal